code/modules/mob/mob.dm

```diff
@@ -254,10 +254,16 @@
 	return TRUE
 
 //This is an UNSAFE proc. It handles situations of timed equips.
-/mob/proc/equip_to_slot_timed(obj/item/W, slot, redraw_mob = 1, permanent = 0, start_loc)
+/mob/proc/equip_to_slot_timed(obj/item/W, slot, redraw_mob = 1, permanent = 0, start_loc, del_on_fail = 0, disable_warning = 0)
 	if(!do_after(src, W.time_to_equip, INTERRUPT_ALL, BUSY_ICON_GENERIC))
 		to_chat(src, SPAN_WARNING("You stop putting on \the [W]!"))
 		return
+	if(!W.mob_can_equip(src, slot, disable_warning)) // we have to do these checks again as circumstances may have changed during the do_after
+		if(del_on_fail)
+			qdel(W)
+		else if(!disable_warning)
```

Comment:
Is there a chance you'd want both the item to be deleted on failure as well as give the warning?

Reply:
no, this code is just copied from /mob/proc/equip_to_slot_if_possible